app/src/main/java/com/example/development/sakaiclientandroid/fragments/AllCoursesFragment.java

```diff
@@ -49,98 +72,134 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 
 
         View view = inflater.inflate(R.layout.fragment_all_courses, null);
+        this.swipeRefreshLayout = view.findViewById(R.id.swiperefresh);
 
 
-        //gets courses from data handler and feeds to list view
-        ExpandableListView sitesListView = view.findViewById(R.id.all_courses_explistview);
-        feedExpandableListData(sitesListView);
-
+        //add the treeview to the layout
+        createTreeView(this.courses);
+        this.swipeRefreshLayout.addView(treeView.getView());
 
-        //expands latest semester's list view by default
-        sitesListView.expandGroup(0);
+        //temporarily disable animations so the animations don't play when the state
+        //is being restored
+        treeView.setDefaultAnimation(false);
```

Comment:
temporarily disables animations so animations don't play when the state is restored. What do you think?

Reply:
I think this is good in the event that the user moves back and forth between fragments, the animations would cause confusion otherwise since it might be unexpected.